pic32/libraries/SPI/SPI.h

```diff
@@ -180,10 +183,10 @@ class SPIClass {
 
         inTransactionFlag = 1;
 #endif
-        pspi->sxCon.clr = (1 << _SPICON_ON);
+        // This causes a nasty spike
+        //pspi->sxCon.clr = (1 << _SPICON_ON);
```

Comment:
I really like all the rest of these changes. But this one I have a question about: here we are going to be changing the baud rate register (BRG) and the SPI Control register (CON), and it looks like the previous code disabled the SPI port (_SPICON_ON cleared), made the changes, then enabled the SPI port (_SPICON_ON set). 

So is it OK to change it so that you never clear the _SPICON_ON bit, but just jam new values into BRG and CON without turning the SPI port off and back on again? I couldn't find anything in the PIC32 reference guide that says this is a bad thing, but every one of their pseudo-code sections disables the port, makes changes to these registers, then re-enables the port.

Maybe a compromise would be to change the code so that the disable and re-enable of the SPI port is ONLY done if the new BRG or CON values are different from the existing values. Then, I would assume, it would happen once at the first transaction, and never after that unless you changed something.